src/components/badge/badge.test.tsx

```diff
@@ -0,0 +1,102 @@
+import React from "react";
+import { render, screen } from "@testing-library/react";
+import Badge from "./badge.component";
+
+const renderComponent = (props = {}) =>
+  render(<Badge {...props}>Button</Badge>);
+
+describe("Badge", () => {
+  it("should render correctly if counter is a number", async () => {
+    renderComponent({ counter: 1 });
+
+    expect(screen.getByText("1")).toBeVisible();
+  });
+
+  it("should render correctly if counter is a string", async () => {
+    renderComponent({ counter: "1" });
+
+    expect(screen.getByText("1")).toBeVisible();
+  });
+
+  it("should render number when counter is between 1 and 99", async () => {
+    renderComponent({ counter: 50 });
+
+    expect(screen.getByText("50")).toBeVisible();
+  });
+
+  it("should render number `99` when counter is higher than 99", async () => {
+    renderComponent({ counter: 101 });
+
+    expect(screen.getByText("99")).toBeVisible();
+  });
+
+  it("should not render badge when counter is `0`", async () => {
+    renderComponent({ counter: 0, onClick: () => {} });
+
+    expect(screen.queryByText("0")).not.toBeInTheDocument();
+    expect(screen.queryByRole("button")).not.toBeInTheDocument();
+  });
+
+  it("should not render badge when counter is not set", async () => {
+    renderComponent({ onClick: () => {} });
+
+    expect(screen.queryByRole("button")).not.toBeInTheDocument();
+  });
+
+  it("should render as a button element when onClick is set", async () => {
+    renderComponent({ counter: 9, onClick: () => {} });
+
+    // investigate why toBeVisible() fails to find "9"
```

Comment:
I wonder if the fact the close icon is also rendered here and includes a pseudo that it's appearing hidden to RTL 🤷.  If we end up needing to do an investigation separate from this ticket then we should raise one and include it in this comment as a FIXMe

Reply:
After some investigation, removing `display: none;` CSS rule on Line 75 in `badge.style.ts` causes the test to pass. Sounds to me like either the pseudoelement @edleeks87 mentioned is causing unintended effects, or JSDOM, or even both. Worth us looking into in the future for sure :+1:

@nuria1110 would you be happy to raise a ticket for this?

Reply:
Raised FE-6575 to investigate this.